azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -195,87 +213,94 @@ def _fetch_goal_state(wire_client):
+                extensions_config = ExtensionsGoalStateFactory.create_from_extensions_config(incarnation, xml_text, self._wire_client)
                 self._history.save_extensions_config(extensions_config.get_redacted_text())
 
-            vm_settings, vm_settings_updated = self._fetch_vm_settings(force_update=force_vm_settings_update)
-
-            if vm_settings is not None:
-                new = " new " if vm_settings_updated else " "
-                logger.info("Fetched{0}vmSettings [HostGAPlugin correlation ID: {1} eTag: {2} source: {3}]", new, vm_settings.hostga_plugin_correlation_id, vm_settings.etag, vm_settings.source)
-                self._extensions_goal_state = vm_settings
-                if vm_settings_updated:
-                    self._history.save_vm_settings(vm_settings.get_redacted_text())
-            else:
-                self._extensions_goal_state = extensions_config
-
             hosting_env_uri = findtext(xml_doc, "HostingEnvironmentConfig")
             xml_text = self._wire_client.fetch_config(hosting_env_uri, self._wire_client.get_header())
-            self._hosting_env = HostingEnv(xml_text)
+            hosting_env = HostingEnv(xml_text)
             self._history.save_hosting_env(xml_text)
 
             shared_conf_uri = findtext(xml_doc, "SharedConfig")
             xml_text = self._wire_client.fetch_config(shared_conf_uri, self._wire_client.get_header())
-            self._shared_conf = SharedConfig(xml_text)
+            shared_conf = SharedConfig(xml_text)
             self._history.save_shared_conf(xml_text)
 
+            certs = None
             certs_uri = findtext(xml_doc, "Certificates")
             if certs_uri is not None:
                 # Note that we do not save the certificates to the goal state history
                 xml_text = self._wire_client.fetch_config(certs_uri, self._wire_client.get_header_for_cert())
-                self._certs = Certificates(xml_text)
+                certs = Certificates(xml_text)
 
-            container = find(xml_doc, "Container")
+            remote_access = None
             remote_access_uri = findtext(container, "RemoteAccessInfo")
             if remote_access_uri is not None:
                 xml_text = self._wire_client.fetch_config(remote_access_uri, self._wire_client.get_header_for_cert())
-                self._remote_access = RemoteAccess(xml_text)
+                remote_access = RemoteAccess(xml_text)
                 self._history.save_remote_access(xml_text)
 
+            self._incarnation = incarnation
+            self._role_instance_id = role_instance_id
+            self._role_config_name = role_config_name
+            self._container_id = container_id
+            self._hosting_env = hosting_env
+            self._shared_conf = shared_conf
+            self._certs = certs
+            self._remote_access = remote_access
+
```

Comment:
We need to do all the network requests and parse the response before updating self; otherwise an exception would lead to a partially updated goal state